opennms-icmp/opennms-icmp-best/src/main/java/org/opennms/netmgt/icmp/best/BestMatchPingerFactory.java

```diff
@@ -85,7 +86,8 @@ private static PingerMatch tryPinger(final Class<? extends Pinger> pingerClass)
 
         try {
             final long timeout = Long.valueOf(System.getProperty("org.opennms.netmgt.icmp.best.timeout", "500"), 10);
-            final Number result = pinger.ping(InetAddressUtils.getLocalHostAddress(), timeout, 0);
+            final InetAddress addr = InetAddressUtils.getLocalLoopbackAddress().orElseThrow(() -> new IllegalStateException("No pingable address found."));
```

Comment:
We could move this out of the `tryPing` loop, and pass a reference to the interface to the function instead.

It may also be worth trying some hard-coded address such as `::1` and `127.0.0.1` as a last ditch effort before falling back to the `NullPinger` as well.